src/mds/MDSMap.cc

```diff
@@ -614,81 +614,87 @@ std::string MDSMap::mds_info_t::human_name() const
   return out.str();
 }
 
-void MDSMap::encode(bufferlist& bl, uint64_t features) const
+void MDSMap::encode(bufferlist& bl, uint64_t features, bool encode_ev) const
 {
-  std::map<mds_rank_t,int32_t> inc;  // Legacy field, fake it so that
-                                     // old-mon peers have something sane
-                                     // during upgrade
+
+  if (!HAVE_FEATURE(features, SERVER_PACIFIC)) {
+    encode_legacy(bl, features);
+    return;
+  }
+
+  // Legacy field, fake it so that old-mon peers have something sane during upgrade
+  std::map<mds_rank_t,int32_t> inc;
+
   for (const auto rank : in) {
     inc.insert(std::make_pair(rank, epoch));
   }
 
   using ceph::encode;
-  if ((features & CEPH_FEATURE_PGID64) == 0) {
-    __u16 v = 2;
-    encode(v, bl);
-    encode(epoch, bl);
-    encode(flags, bl);
-    encode(last_failure, bl);
-    encode(root, bl);
-    encode(session_timeout, bl);
-    encode(session_autoclose, bl);
-    encode(max_file_size, bl);
-    encode(max_mds, bl);
-    __u32 n = mds_info.size();
-    encode(n, bl);
-    for (map<mds_gid_t, mds_info_t>::const_iterator i = mds_info.begin();
-	i != mds_info.end(); ++i) {
-      encode(i->first, bl);
-      encode(i->second, bl, features);
-    }
-    n = data_pools.size();
-    encode(n, bl);
-    for (const auto p: data_pools) {
-      n = p;
-      encode(n, bl);
-    }
+  ENCODE_START(6, 4, bl);
+  encode(epoch, bl);
+  encode(flags, bl);
+  encode(last_failure, bl);
+  encode(root, bl);
+  encode(session_timeout, bl);
+  encode(session_autoclose, bl);
+  encode(max_file_size, bl);
+  encode(max_mds, bl);
+  encode(mds_info, bl, features);
+  encode(data_pools, bl);
+  encode(cas_pool, bl);
 
-    int32_t m = cas_pool;
-    encode(m, bl);
-    return;
-  } else if ((features & CEPH_FEATURE_MDSENC) == 0) {
-    __u16 v = 3;
-    encode(v, bl);
-    encode(epoch, bl);
-    encode(flags, bl);
-    encode(last_failure, bl);
-    encode(root, bl);
-    encode(session_timeout, bl);
-    encode(session_autoclose, bl);
-    encode(max_file_size, bl);
-    encode(max_mds, bl);
-    __u32 n = mds_info.size();
-    encode(n, bl);
-    for (map<mds_gid_t, mds_info_t>::const_iterator i = mds_info.begin();
-	i != mds_info.end(); ++i) {
-      encode(i->first, bl);
-      encode(i->second, bl, features);
-    }
-    encode(data_pools, bl);
-    encode(cas_pool, bl);
+  __u16 ev = 9;
+  encode(ev, bl);
+  {
+    CompatSet _compat;
+    encode(_compat, bl);
+  }
+  encode((int64_t)0, bl);
```

Comment:
What does this correspond to?

Reply:
metadata_pool. See https://github.com/ceph/ceph/pull/33835/files#diff-c4d6a907104e07425dc624f7a66f572aR676